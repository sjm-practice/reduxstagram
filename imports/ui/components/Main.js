// @flow
import React, {
  Component,
} from "react";
import { Link } from "react-router";

type IProps = {
  children: React.Element<*>,
};

// eslint-disable-next-line react/prefer-stateless-function
class Main extends Component {
  props: IProps;

  render() {
    const { children, ...otherProps } = this.props;
    const newChild = React.cloneElement(children, otherProps);

    return (
      <div>
        <h1>
          <Link to="/">
            Reduxstagram
          </Link>
        </h1>
        {newChild}
      </div>
    );
  }
}

export default Main;
