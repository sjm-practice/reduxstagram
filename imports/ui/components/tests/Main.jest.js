// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import renderer from "react-test-renderer";
import Main from "../Main";

describe("<Main />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

