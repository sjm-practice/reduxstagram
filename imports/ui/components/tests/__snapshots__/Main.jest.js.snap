exports[`<Main /> matches render snapshot 1`] = `
<div>
  <h1>
    <a
      onClick={[Function]}
      style={Object {}}>
      Reduxstagram
    </a>
  </h1>
  <p>
    test child element
  </p>
</div>
`;
